frontend/model/state.js

```diff
@@ -90,7 +158,7 @@ sbp('sbp/selectors/register', {
       // $FlowFixMe[incompatible-call]
       Vue.set(state, 'reverseNamespaceLookups', Object.fromEntries(Object.entries(state.namespaceLookups).map(([k, v]: [string, string]) => [v, k])))
     }
-    (() => {
+    contractUpdate(state, (state: Object, contractIDHints: ?string[]) => {
```

Comment:
I asked Claude if it could make the code from line 170 to 182 more readable, and here's what it suggested:

```js
Object.entries(state[ourIdentityContractId].groups)
  .filter(([groupID, { hasLeft }]) => {
    return !hasLeft && 
           state[groupID]?.chatRooms && 
           (!Array.isArray(contractIDHints) || contractIDHints.includes(groupID))
  })
  .map(([groupID]) => {
    // $FlowFixMe[incompatible-use]
    const chatRooms = state[groupID].chatRooms
    const needsUpgrade = Object.values(chatRooms)
      .flatMap(({ members }) => Object.values(members))
      .some(member => 
        member.status === PROFILE_STATUS.ACTIVE && member.joinedHeight == null
      )

    return needsUpgrade ? groupID : null
  })
  .filter(Boolean)
```

If this accomplishes the same thing, do you think it's safe to replace it with this?

Reply:
Maybe, I'd need to see if they do the same thing or not